packages/lib-classifier/src/components/Classifier/components/SubjectViewer/components/SingleVideoViewer/SingleVideoViewerContainer.js

```diff
@@ -23,270 +23,182 @@ const DrawingLayer = styled.div`
   cursor: default;
 `
 
-class SingleVideoViewerContainer extends React.Component {
-  constructor() {
-    super()
-
-    this.videoViewer = React.createRef()
-    this.transformLayer = React.createRef()
-
-    this.state = {
-      vid: '',
-      isPlaying: false,
-      playbackRate: 1,
-      played: 0,
-      duration: 0,
-      isSeeking: false,
-      clientWidth: 0,
-      naturalWidth: 0,
-      naturalHeight: 0
+function SingleVideoViewerContainer({
+  enableInteractionLayer = true,
+  onError = () => true,
+  onReady = () => true,
+  onKeyDown = () => true,
+  subject
+}) {
+  const [clientWidth, setClientWidth] = useState(0)
+  const [duration, setDuration] = useState(0)
+  const [isPlaying, setIsPlaying] = useState(false)
+  const [isSeeking, setIsSeeking] = useState(false)
+  const [playbackRate, setPlaybackRate] = useState(1)
+  const [timeStamp, setTimeStamp] = useState(0)
+  const [videoHeight, setVideoHeight] = useState(0)
+  const [videoWidth, setVideoWidth] = useState(0)
+  const [videoSrc, setVideoSrc] = useState('')
+
+  const interactionLayerSVG = useRef()
+  const playerRef = useRef()
+  const transformLayer = useRef()
+
+  const { t } = useTranslation('components')
+
+  /* ==================== load subject ==================== */
+
+  useEffect(() => {
+    if (subject) {
+      setVideoSrc(Object.values(subject.locations[0])[0])
     }
```

Comment:
```suggestion
  const videoSrc = subject ? Object.values(subject.locations[0])[0] : null
```

You can assign `videoSrc` directly here, which gets rid of an extra render.

https://github.com/zooniverse/front-end-monorepo/blob/641905adfffad4a0e06953872d55430b2bb57d86/packages/lib-classifier/src/components/Classifier/components/SubjectViewer/components/SingleImageViewer/SingleImageViewerContainer.js#L33-L37